scripts/table_diff/changes/changes.go

```diff
@@ -47,13 +48,35 @@ func parseColumnChange(line string) (name string, dataType string, pk bool) {
 	return cleanName, result["dataType"], result["name"] != cleanName
 }
 
+func parsePKChange(line string) (names []string) {
+	for _, match := range pkRegex.FindAllStringSubmatch(line, -1) {
+		if len(match) != 2 {
+			continue
+		}
+		for _, part := range strings.Split(match[1], ", ") {
+			names = append(names, strings.Trim(part, "*"))
+		}
+	}
+	return
+}
+
 func getColumnChanges(file *gitdiff.File, table string) (changes []change) {
 	addedColumns := make(map[string]column)
 	deletedColumns := make(map[string]column)
+	var addedPK, deletedPK []string
 	for _, fragment := range file.TextFragments {
 		for _, line := range fragment.Lines {
 			name, dataType, pk := parseColumnChange(line.Line)
 			if name == "" || dataType == "" {
+				pkChanges := parsePKChange(line.Line)
```

Comment:
Can't we move this up just at the beginning of the loop here:
https://github.com/cloudquery/cloudquery/pull/6021/files#diff-6910d6ee0a1eadb1d8f2122f804f6ab3ef6e4c6d040b631527d79e73b4678c50R69
to avoid the nesting?